src/binder/bind/read/bind_in_query_call.cpp

```diff
@@ -22,47 +25,71 @@ std::unique_ptr<BoundReadingClause> Binder::bindInQueryCall(const ReadingClause&
+        childrenTypes.push_back(child->getDataType());
     }
+    std::unique_ptr<BoundReadingClause> boundReadingClause;
+    expression_vector columns;
     auto catalogSet = clientContext->getCatalog()->getFunctions(clientContext->getTx());
-    auto functionEntry = BuiltInFunctionsUtils::getFunctionCatalogEntry(clientContext->getTx(),
+    auto entry = BuiltInFunctionsUtils::getFunctionCatalogEntry(clientContext->getTx(),
         functionName, catalogSet);
-    if (functionEntry->getType() != CatalogEntryType::TABLE_FUNCTION_ENTRY) {
-        throw BinderException(stringFormat("{} is not a table function.", functionName));
-    }
-    auto func = BuiltInFunctionsUtils::matchFunction(functionName, inputTypes, functionEntry);
-    tableFunction = *func->constPtrCast<TableFunction>();
-    for (auto i = 0u; i < children.size(); ++i) {
-        auto parameterTypeID = tableFunction.parameterTypeIDs[i];
-        ExpressionUtil::validateDataType(*children[i], parameterTypeID);
-    }
-    auto bindInput = function::TableFuncBindInput();
-    bindInput.inputs = std::move(inputValues);
-    auto bindData = tableFunction.bindFunc(clientContext, &bindInput);
-    expression_vector columns;
-    for (auto i = 0u; i < bindData->columnTypes.size(); i++) {
-        columns.push_back(createVariable(bindData->columnNames[i], bindData->columnTypes[i]));
+    auto func = BuiltInFunctionsUtils::matchFunction(functionName, childrenTypes, entry);
+    switch (entry->getType()) {
+    case CatalogEntryType::TABLE_FUNCTION_ENTRY: {
+        // Bind CALL TABLE_FUNCTION.
+        std::vector<Value> inputValues;
+        for (auto& param : children) {
+            ExpressionUtil::validateExpressionType(*param, ExpressionType::LITERAL);
+            auto literalExpr = param->constPtrCast<LiteralExpression>();
+            inputValues.push_back(literalExpr->getValue());
+        }
+        auto tableFunc = func->constPtrCast<TableFunction>();
+        for (auto i = 0u; i < children.size(); ++i) {
+            auto parameterTypeID = tableFunc->parameterTypeIDs[i];
+            ExpressionUtil::validateDataType(*children[i], parameterTypeID);
+        }
+        auto bindInput = function::TableFuncBindInput();
+        bindInput.inputs = std::move(inputValues);
+        auto bindData = tableFunc->bindFunc(clientContext, &bindInput);
+        for (auto i = 0u; i < bindData->columnTypes.size(); i++) {
+            columns.push_back(createVariable(bindData->columnNames[i], bindData->columnTypes[i]));
+        }
+        auto offset = expressionBinder.createVariableExpression(*LogicalType::INT64(),
+            std::string(InternalKeyword::ROW_OFFSET));
+        boundReadingClause = std::make_unique<BoundInQueryCall>(*tableFunc, std::move(bindData),
+            std::move(offset), std::move(columns));
+    } break;
+    case CatalogEntryType::GDS_FUNCTION_ENTRY: {
+        // Bind CALL ALGORITHM_FUNCTION.
+        auto gdsFunc = *func->constPtrCast<GDSFunction>();
```

Comment:
I think you're doing an implicit copy here. Is that expected? You might want `auto& gdsFunc` instead.

Reply:
This is expected.